File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/services/BrokerService.java
package com.chaincuerealestate.chaincuerealestate.services;

import com.chaincuerealestate.chaincuerealestate.domains.Broker;
import com.chaincuerealestate.chaincuerealestate.repositories.BrokerRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class BrokerService implements BrokerServiceI {
    private final BrokerRepository brokerRepository;

    public Broker save(String email) {
        Broker broker = Broker.create(email);
        return brokerRepository.save(broker);
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/services/BrokerService.kt
package com.example.chaincuerealestate.services

import com.example.chaincuerealestate.domains.Broker
import com.example.chaincuerealestate.repositories.BrokerRepository
import org.springframework.stereotype.Service

@Service
class BrokerService(private val brokerRepository: BrokerRepository) : BrokerServiceI {

    fun save(email: String): Broker {
        val broker = Broker.create(email)
        return brokerRepository.save(broker)
    }
}
